src/page/AddMenu/components/MenuCategory/index.tsx

```diff
@@ -1,91 +1,145 @@
 import React, { useState } from 'react';
 import { ReactComponent as GearIcon } from 'assets/svg/mystore/gear.svg';
 import useMediaQuery from 'utils/hooks/useMediaQuery';
+import useAddMenuStore from 'store/addMenu';
+import useMyShop from 'query/shop';
 import cn from 'utils/ts/className';
 import styles from './MenuCategory.module.scss';
 
 interface MenuCategory {
   id: number;
   name: string;
 }
+interface MenuCategoryProps {
+  isComplete: boolean;
+}
 
-export default function MenuCategory() {
+export default function MenuCategory({ isComplete }:MenuCategoryProps) {
   const { isMobile } = useMediaQuery();
-  const [menuCategories/* , setMenuCategories */] = useState<MenuCategory[]>([
-    { id: 1, name: '이벤트 메뉴' },
-    { id: 2, name: '대표 메뉴' },
-    { id: 3, name: '사이드 메뉴' },
-    { id: 4, name: '세트 메뉴' },
-  ]);
+  const { shopData } = useMyShop();
+  const { setCategoryIds } = useAddMenuStore();
   const [selectedCategories, setSelectedCategories] = useState<MenuCategory[]>([]);
   const appendSelectCategory = (category: MenuCategory) => {
     if (selectedCategories.some((alreadySelected) => alreadySelected.id === category.id)) {
-      setSelectedCategories(
-        selectedCategories.filter((alreadySelected) => alreadySelected.id !== category.id),
+      const newSelected = selectedCategories.filter(
+        (alreadySelected) => alreadySelected.id !== category.id,
```

Comment:
```suggestion
const appendSelectCategory = (category: MenuCategory) => {
  const newSelected = selectedCategories.some((c) => c.id === category.id)
    ? selectedCategories.filter((c) => c.id !== category.id)
    : [...selectedCategories, category];

  setSelectedCategories(newSelected);
  setCategoryIds(newSelected.map((cat) => cat.id));
};
```
겹치는 연산이 있어서 간단하게 줄일 수 있을 거 같아요!

Reply:
수정 완료했습니다.!
6721660dfba0348b49d38f06d98204e890f0a991